src/transport.ts

```diff
@@ -69,6 +74,15 @@ export class Transport {
       if (this.apiKey !== undefined) {
         lam.addEnvironment(API_KEY_ENV_VAR, this.apiKey);
       }
+      const isNode = runtimeLookup[lam.runtime.name] === RuntimeType.NODE
```

Comment:
It looks like if you are using synchronous metrics with Node, we will accept the `DD_API_KEY_SECRET_ARN` environment variable in the `validateProps()` function, but then ignore it here. Also, we log a debug log line, which is only visible when debug logs are turned on, but otherwise there is no indication anything is wrong.

What do you want the user experience to be here?

Reply:
The main problem is that in `validateProps()` I don't have the runtime of the function, therefore I am not able to check if it will be possible to proceed. Then, I do it in the `transport.ts`. Will try to think of a way to let the user know we are ignoring the secret ARN.

Reply:
Ok, so as suggested, I will change it to an error, since we want the user to know that if he sets the secret ARN and is using sync metrics and a node runtime, it will not work.